### DISCUSSION TITLE
Can't Load Previous User Data After Install Mishap

### DISCUSSION THREAD

**Discussion by iioniis** [nicotine-plus/nicotine-plus#3030] [Q&A] [answered]
I was downloading the new run and I accidentally ran the Silicone installer (I have an Intel Mac).  I ran it and it wouldn't open obviously and once I realized what I did, I pulled my '.local/share/nicotine/' folder and my '/.config/nicotine/' folder, uninstalled the Silicone version, then ran the Intel installer after putting those folders back and now Nicotine won't open like it wouldn't when the Silicone version was installed.  Can I save my user data?  I feel like an idiot.  Any help would be greatly appreciated.  Thank you.

**Comment by slook** [comment]
> ran the Intel installer after putting those folders back

Try do it the other way around...

Put your local user data and config aside (somewhere safe) for the time being, then uninstall, reinstall Nicotine+ fresh (i.e. _**before**_ restoring local user data and config) so you successfully get to the Setup Assistant, then close Nicotine+ and finally restore your data and config.

While Nicotine+ is uninstalled, reboot your system so that any left over cached Python files are removed.

**Reply by iioniis** [reply]
Here is what I did after reading your suggestions:

1. Put my 'config/nicotine' and 'share/nictoine' folders somewhere safe.
2. Uninstalled Nicotine
3. Rebooted computer
4. Installed Nicotine
5. Opened Nicotine and when the setup window appeared I quit Nicotine and put my config files back in 'root/users/user/.config/nicotine' and my data back in /root/users/user/.local/share/nicotine'.
6. Opened Nicotine and it just hangs there with the dock icon bouncing, not responding until I have to force quit.

Do you think it could just be loading everything and scanning my shares before becoming operational?  maybe the share directory is different or the computer login user was my old one?

**Reply by iioniis** [reply]
UPDATE:  Cleaned both of those folders out completely and let it boot into Nicotine so as to create these files again.  Removed all files from share/nicotine and replaced with mine, didn't freeze but didn't load any of my stuff.  Left the .config file alone.

**Reply by slook** [reply]
> it could just be loading everything and scanning my shares before becoming operational?

That is possible. The main window is supposed to draw before starting rescan, but I have observed this not working as intended during first run (when there is a full rebuild, not just a quick rescan) on some systems.

> the computer login user was my old one?

The only thing that might be relevant is if the username is in any folder paths, then these will need to be manually changed if you are migrating to a different system/account.

Perhaps there could be a cascade of folder access permission errors to get through during startup, that the program gets caught up on before having opportunity to draw the main window.

We would appreciate further testing, if possible. Can you try placing your user data and then running `nicotine` from the terminal, so that you can observe status log messages?

The `--config` and `--user-data` line arguments might be helpful (rather than having to copy files) for troubleshooting purposes, see...

```
$ ./nicotine --help
usage: nicotine [-h] [-c file] [-u dir] [-s] [-b ip] [-l port] [-r] [-n] [-v]

Graphical client for the Soulseek peer-to-peer network

optional arguments:
  -h, --help                 show this help message and exit
  -c file, --config file     use non-default configuration file
  -u dir, --user-data dir    alternative directory for user data and plugins
...
```

Although, do still keep a seperate backup of your stuff somewhere safe if you are going to test this issue further.

**Reply by iioniis** [reply]
I will do this and get back to you.  I went back and did a clean install of 3.3.4 (which I haven used in a little bit), started with putting my data files and config files in at the same time from a backup about a month ago and it worked.  But now I just tried installing the newest run over 3.3.4 and it's doing the thing again.  I'm going to let it go for a bit then open in terminal.

**Reply by iioniis** [reply]
Here is what I got:

`[2024-06-08 08:34:18] Loading Python 3.12.3
[2024-06-08 08:34:18] Loading Nicotine+ 3.3.5.dev1
[2024-06-08 08:34:18] Loading GTK 4.14.4
usage: nicotine [-h] [-c file] [-u dir] [-s] [-b ip] [-l port] [-r] [-n] [-v]
nicotine: error: unrecognized arguments: --multiprocessing-fork tracker_fd=16 pipe_handle=20
`

**Reply by mathiascode** [reply]
This is only an issue in 3.3.5.dev1 then? Probably a regression in cx_Freeze (used to package the Windows and macOS builds).

**Reply by iioniis** [reply]
I guess so.  I installed 3.3.4 and it’s fine. In the beginning I do believe I still installed the Silicone version. I don’t want to recreate it but 3.3.5 dev 1 definitely is what’s doing it.

**Reply by mathiascode** [reply, marked as answer]
The issue should be fixed in the latest development build. Please ensure no Nicotine+ processes are running on the system, and try the new build.

**Reply by iioniis** [reply]
It works!  Thank you.
